### PR TITLE
Add output of debug:messenger information on Messenger doc

### LINKED ISSUE #13310: [Messenger] Show message & handler(s) class description in debug:messen…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#35978
| PR author(s) | @ogizanagi
| Merged in    | 5.1-dev

### PR DESCRIPTION
**Pull request by jpjoao**
fixes #13310 :
* Adds code for a dummy message `MyMessage` to support Examples;
* Use MyMessage on handler example so `debug:messenger` output makes sense;
* Adds a quick description to what a Message can be as this was missing;

### CODE REVIEW

**Comment by jpjoao** [comment]
I was wondering if it makes sense to change all the examples on that page to cope with the `NewOrder` message that is more interesting and could be more useful for explaining the component.
I could either do it in a separated PR or refactor this one.

**Comment by javiereguiluz** [comment]
@Nyholm could you please review this proposal? Thanks!

File: components/messenger.rst [hunk 1]
```diff
@@ -68,6 +68,34 @@ Concepts
    to use for transport, markers identifying a received message or any sort of
    metadata your middleware or transport layer may use.
 
+**Message**
+   There is no specific requirement for a message, it can be any kind of object except it
+   should be serializable and unserializable by a Symfony Serializer instance.
```
Comment (Koc) [inline components/messenger.rst:73]: Actually not. Native serialization mechanism [is used by default](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L1222-L1224)

**Comment by Nyholm** [comment]
I like the part of "Debugging the Messenger". But Im not sure the other changes are needed.

File: components/messenger.rst [hunk 2]
```diff
@@ -219,13 +248,13 @@ you can create your own message sender::
...
 
-    use App\Message\ImportantAction;
+    use App\Message\MyMessage;
     use Symfony\Component\Mailer\MailerInterface;
     use Symfony\Component\Messenger\Envelope;
     use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
     use Symfony\Component\Mime\Email;
 
-    class ImportantActionToEmailSender implements SenderInterface
+    class MyMessageToEmailSender implements SenderInterface
```
Comment (OskarStark) [inline components/messenger.rst:257]: I would keep the old name

**Comment by Nyholm** [comment]
@jpjoao If you split this PR into 2 PRs then we can easily merge the fix for https://github.com/symfony/symfony-docs/issues/13310

The other changes are a bit more discussions around so it might take longer time for those changes to be merged.

**Comment by Nyholm** [comment]
Friendly ping

**Comment by wouterj** [comment]
Hi! Thank you for initiating this PR! Unfortunately, there hasn't been much activity on this PR for the past months. So I'm going to close this PR. Feel free to comment (to reopen) or open a new PR with the feedback applied.

As for #13310, I could not see the relation with this PR (the PHPdoc description is not shown). I've created https://github.com/symfony/symfony-docs/pull/14567 to add a small tip about the feature mentioned in the issue.

**Pull request closed without merging**